tools/integration_tests/readonly/copy_object_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provides integration tests for file operations with --o=ro flag set.
+package readonly_test
+
+import (
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"syscall"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/setup"
+)
+
+// Copy srcFile in testBucket/Test/b/b.txt destination.
+func checkIfFileCopyFailed(srcFilePath string, t *testing.T) {
+	source, err := os.OpenFile(srcFilePath, syscall.O_DIRECT, setup.FilePermission_0600)
+	if err != nil {
+		t.Errorf("Error in the opening file: %v", err)
+	}
+
+	// Checking if destination object exist.
+	copyFile := path.Join(setup.MntDir(), DirectoryNameInTestBucket, SubDirectoryNameInTestBucket, FileNameInSubDirectoryTestBucket)
```

Comment:
Why is it necessary to have destination file exist for copy to work here ? We don't need in linux cp operation.

Reply:
As it is readonly file system we can't create the file if it does not exist.

Reply:
In that case, I would suggest to add a comment that cp without destination file creates a destination file and create workflow is already covered separately.